### PR TITLE
PCHR-2113: hrvisa activity handling fixes

### PR DESCRIPTION
**Pull request by davialexandre**
One of the features of the hrvisa extension is to check for expired visas and, in case this is a required document and the existing one is expired, it creates Activities to "notify" the user that it has to be updated.

In order to do this, it first checks if such Activity already exists (to avoid spamming the user with duplicated activities) and it does that by looking for activities where the Contact of the expired visa is the target. This check was done by inspecting the `targets` property returned by the API, but since the changes introduced by https://github.com/civicrm/civicrm-core/pull/9841, `targets` is not returned anymore and the code was failing.

To fix this, I first thought about replacing `targets` with something else, but then I realized that this check is not required at all. Looking at the code, there was this comment: 

> note : using filter 'activity_type_id' in combination with 'contact_id' filter doesn't work

It means, the check was only added because it was not possible, by the time the code was created back in 2013, to get an activity by both the type and the contact. Fortunately, this is not the case anymore. The API works just fine now, so I completely remove the check, which fixed the error.

Finally, I did some cleanup to fix some typos and remove the dependency on CiviUnitTestCase, which we're not using anymore on the new tests.

### CODE REVIEW

**Comment by mickadoo** [comment]
Just to check, but the PCHR-2113 issue number is used in [multiple](https://github.com/civicrm/civihr/pull/1729) [pull](https://github.com/civicrm/civihr/pull/1710) [requests](https://github.com/civicrm/civihr/pull/1730) - this is intentionally splitting the ticket into smaller PRs?

File: hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php [hunk 1]
```diff
@@ -20,7 +20,7 @@ public function setUpHeadless() {
       ->apply();
   }
 
-  function setUp() {
+  public function setUp() {
```
Comment (mickadoo) [inline hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php]: +1 for visibility on all methods

File: hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php [hunk 2]
```diff
@@ -37,30 +37,27 @@ function setUp() {
...
     $params = array(
       'weight' => 1,
       'label' => 'Visa Expiration',
       'filter' => 0,
       'is_active' => 1,
       'is_default' => 0,
     );
-    $result = civicrm_api3('activity_type', 'create', $params);
+    civicrm_api3('activity_type', 'create', $params);
     return TRUE;
```
Comment (mickadoo) [inline hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php]: I've seen a mix of both, do you think we should have a newline before `return`

File: hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php [hunk 3]
```diff
@@ -145,26 +144,23 @@ function testSyncSncenario2() {
     $this->assertEquals(CRM_Core_OptionGroup::getValue('activity_status', 'Cancelled', 'name'), $activity['values'][$activity['id']]['status_id'], 'in line ' . __LINE__ . ' Status of \'Visa Expiration\' activity should be \'Cancelled\' but wrongly is ' . $activity['values'][$activity['id']]['status_id']);
   }
 
-  function _getTargetContactActivity($contactId) {
+  private function _getTargetContactActivity($contactId) {
```
Comment (mickadoo) [inline hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php]: If we're making this private can we remove the underscore in front of it?

File: hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php [hunk 4]
```diff
@@ -83,7 +80,7 @@ function testSyncScenario1() {
     // sync activity with contact of above details
     CRM_HRVisa_Activity::sync($cid);
 
-    // calling a common function for getting acivity a particular target contact and acitvity type
```
Comment (mickadoo) [inline hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php]: two different ways to misspell activity in the one sentence, impressive :-D

**Comment by davialexandre** [comment]
I decided for submitting separate PRs for a few reasons:
- To have smaller PRs
- The issues were caused by different changes and affected different parts of the system
- https://github.com/civicrm/civihr/pull/1730 is a fix for CiviHR 1.7, and has to be merged into a different branch

File: hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php [hunk 5]
```diff
@@ -2,25 +2,27 @@
...
 use Civi\Test\TransactionalInterface;
+use CRM_HRCore_Test_Fabricator_Contact as ContactFabricator;
 
 /**
  * Class CRM_HRVisa_ActivityTest
  *
  * @group headless
  */
-class CRM_HRVisa_ActivityTest extends CiviUnitTestCase implements HeadlessInterface , TransactionalInterface {
+class CRM_HRVisa_ActivityTest extends PHPUnit_Framework_TestCase implements HeadlessInterface , TransactionalInterface {
```
Comment (mickadoo) [inline hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php]: I like the idea of using vanilla PHPUnit framework tests

File: hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php [hunk 6]
```diff
@@ -145,42 +131,40 @@ function testSyncSncenario2() {
...
-    $import->run(
-      CRM_Extension_System::singleton()->getMapper()->keyToBasePath('org.civicrm.hrvisa')
-      . '/xml/auto_install.xml'
-    );
-    // this had to be done as demographics consists of is_visa_required field (used in unit test)
-    $import->run(
-      CRM_Extension_System::singleton()->getMapper()->keyToBasePath('org.civicrm.hrdemog')
-      . '/xml/auto_install.xml'
-    );
+  private function createLoggedInUser() {
```
Comment (mickadoo) [inline hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php]: This method is nice, clean and descriptive. I wonder would it be worth putting in a trait to be used in other tests?

Comment (davialexandre) [inline hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php]: That's a great a idea. In fact, in L&A we do have a trait that does almost the same thing: https://github.com/civicrm/civihr/blob/la-milestone-3/uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/helpers/SessionHelpersTrait.php

I even thought about adding that to this test, but then I found a few reasons to not do it (for now):
- That trait it's on the 1.7 branch, so I would have to copy it to the 1.6 branch in order to be able to use it here
- It works on a slightly different way than the one we have in SessionHelpersTrait
- I could create a new trait just for this test, but then I would have 2 different traits doing almost the same thing, which is not a good thing

So, my idea is to wait for this until we have 1.7 merged into staging. Also, once we have Jenkins, we'll have to fix and improve a lot of things on the existing old tests. One of these things we'll definitely be extracting this to a generic trait that can be used both by hrvisa and L&A

Comment (mickadoo) [inline hrvisa/tests/phpunit/CRM/HRVisa/ActivityTest.php]: Sounds reasonable. If you're merging 1.6 into 1.7 as you're going along you might consider merging there two traits (unifying the functionality) but leaving it till later sounds fine too. I'm sure there are a lot of other things that can be cleaned up if we spend some time focusing on tests.

**Pull request merged**